src/Model/TransactionalManagerInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Sonata\PageBundle\Model;
+
+interface TransactionalManagerInterface
```

Comment:
This interface need to be introduce in sonata-doctrine-extensions instead.

This is not a PR for this bundle.

Reply:
Then I'll introduce in `sonata-doctrine-extensions` and here correct?

Reply:
Just in sonata-doctrine-extension ; there is no need to duplicate this here.

Reply:
The code will be
```
if ($this->snapshotManager instanceof TransactionalManagerInterface) {
            $this->snapshotManager->beginTransaction();
        }
```

Reply:
Ok, But I can't depends the last version of `sonata-doctrine-extension`, because it forces to use php 8

Reply:
> The code will be
> 
> ```
> if ($this->snapshotManager instanceof TransactionalManagerInterface) {
>             $this->snapshotManager->beginTransaction();
>         }
> ```

and how do I do in the interface?

Reply:
> Ok, But I can't depends the last version of `sonata-doctrine-extension`, because it forces to use php 8
> 
> Edit 1: Or can I add `^7.4 | ^8.0` in composer.json?

I see two options:
- Introducing the Interface in 2.x and 3.x of sonata-doctrine-extensions.
- Just introducing the interface in 3.x and in SonataPageBundle using
```
// TODO: Use TransactionalManagerInterface check instead when drop doctrine-extension 2
if ($this->snapshotManager instanceof BaseEntityManager) {
    $this->snapshotManager->getEntityManager()->beginTransaction();
}
```

This might be ok to introduce the interface in 2.x ; WDYT @jordisala1991 ?

Reply:
Ok question: Can we introduce this in 1.x and depends the 1.x version 👀

Reply:
> Ok question: Can we introduce this in 1.x and depends the 1.x version 👀
> 
> and remove the support for 2.x: [`https://github.com/sonata-project/SonataPageBundle/blob/4.x/composer.json#L30`](https://github.com/sonata-project/SonataPageBundle/blob/4.x/composer.json#L30)
> 
> when we drop the 7.4 support we can support 2.x again, what do you think 👀

No need to remove support for 2.x if you introduce the interface in 1.x.

You'll support `^1.18 || ^2.1` because of both of these version will have the interface

Reply:
🤯 you are right!

Then I'll open the PR to 1.x ok?

Reply:
Done: https://github.com/sonata-project/sonata-doctrine-extensions/pull/443